samples/nrf9160/lwm2m_client/src/ui/led_pwm.c

```diff
@@ -152,7 +153,7 @@ static void led_update(struct led *led)
 		int32_t next_delay =
 			led->effect->steps[led->effect_step].substep_time;
 
-		k_delayed_work_submit(&led->work, K_MSEC(next_delay));
+		k_work_schedule(&led->work, K_MSEC(next_delay));
```

Comment:
This code seems like it's similar to the version discussed at https://github.com/nrfconnect/sdk-nrf/pull/4454#discussion_r630955328, though it doesn't have the potential null pointer dereference.  Here I notice that use of `schedule` means it delays the change in effect until the work is next executed, if a previous effect was active.

Maybe this LED functionality should be pulled out into a library that the samples (and applications) can use.  (Or maybe not; I'm only looking at the diffs here.)